lib/strategy.ts

```diff
@@ -0,0 +1,93 @@
+import { HeaderIndexes, DatasetMapping } from "./models";
+
+
+export interface SortingStrategy {
+    doSort(data: string[], datasetMapping?: DatasetMapping, headerIndexes?: HeaderIndexes, getType?: (row: string[], headerIndexes: HeaderIndexes) => string): string[];
+    toggleSortOrder(): void;
+}
+
+export enum SortState {
+    Off = 0,
+    Ascending = 1,
+    Descending = 2
+}
+
+abstract class SortingStrategyBase implements SortingStrategy {
+    public sortState: SortState = SortState.Ascending;
+
+    public toggleSortOrder(): void {
+        this.sortState = (this.sortState + 1) % 3;
+    }
+
+    abstract doSort(data: string[], datasetMapping: DatasetMapping, headerIndexes: HeaderIndexes, getType?: (row: string[], headerIndexes: HeaderIndexes) => string): string[];
+}
+
+export class CdeSortingFilter extends SortingStrategyBase {
+    public doSort(data: string[], datasetMapping: DatasetMapping, headerIndexes: HeaderIndexes): string[] {
+        if (this.sortState === SortState.Off) {
+            return Object.keys(datasetMapping);
+        }
+
+        const sortedData = data.slice();
+
+        sortedData.sort((a, b) => {
+            const rowA = datasetMapping[a];
+            const rowB = datasetMapping[b];
+            let comparisonResult = rowA[headerIndexes.preciseAbbreviation].localeCompare(rowB[headerIndexes.preciseAbbreviation]);
```

Comment:
Suggestion (non-blocking): You can use https://github.com/MetaCell/cde-mapper/blob/develop/lib/helpers/getters.ts#L7